src/main/java/lotto/domain/LottoMachine.java

```diff
@@ -1,21 +1,97 @@
 package lotto.domain;
 
+import lotto.exception.LottoPurchaseAmountException;
+
+import java.math.BigDecimal;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static lotto.domain.Number.MAX_NUMBER;
+import static lotto.domain.Number.MIN_NUMBER;
+import static lotto.utils.RandomNumberUtils.generateRandomNumbers;
 
 public class LottoMachine {
 
+    public static final int LOTTO_SIZE = 6;
+    public static final Money GAME_PRICE = new Money(BigDecimal.valueOf(1000));
+
     public List<LottoNumber> getLottoList(Money money) {
-        return generateLotto(money.getPurchaseCount());
+        validationMoney(money);
+        int autoCount = getPurchaseCount(money, GAME_PRICE);
+        return generateAutoLotto(autoCount).stream()
+                .collect(Collectors.toList());
+    }
+
+    public List<LottoNumber> getLottoList(Money money, List<List<Integer>> activeNumberList) {
+        int activeCount = activeNumberList.size();
+        int autoCount = getPurchaseCount(money, GAME_PRICE) - activeCount;
+
+        return Stream.concat(generateActiveLotto(activeNumberList).stream(), generateAutoLotto(autoCount).stream())
+                .collect(Collectors.toList());
+    }
+
+    public LottoNumber getLottoNumber(List<Integer> numbers) {
+        return new LottoNumber(numbers.stream()
+                .map(m -> Number.of(m))
+                .collect(Collectors.toList()));
+    }
+
+    public Number getBonusNumber(int bonusNumber) {
+        return Number.of(bonusNumber);
+    }
+
+    private int getPurchaseCount(Money purchaseAmount, Money gamePrice) {
+        return purchaseAmount.getMoney().divide(gamePrice.getMoney()).intValue();
```

Comment:
Money 객체에서 데이터를 꺼내 계산하시는 부분은 Money 클래스를 객체가 아닌 데이터의 저장소로 바라보는 것입니다 😢 
디미터 법칙에 대해 알아보시면 좋을 것 같아요

https://johngrib.github.io/wiki/law-of-demeter/

```suggestion
        return purchaseAmount.divide(gamePrice);
```